packages/rlc-common/src/metadata/buildTsConfig.ts

```diff
@@ -4,39 +4,65 @@
 import { Project } from "ts-morph";
 import { RLCModel } from "../interfaces.js";
 
-const restLevelTsConfigInAzureSdkForJs: Record<string, any> = {
-  extends: "../../../tsconfig.package",
-  compilerOptions: {
-    outDir: "./dist-esm",
-    declarationDir: "./types"
-  },
-  include: ["src/**/*.ts"]
+const restLevelTsConfigInAzureSdkForJs: (
+  model: RLCModel
+) => Record<string, any> = function (model: RLCModel) {
+  if (model.options?.moduleKind === "esm") {
+    return {
+      extends: "../../../tsconfig.package",
+      compilerOptions: {
+        module: "NodeNext",
+        moduleResolution: "NodeNext",
+        rootDir: "."
+      },
+      include: [
+        "./src/**/*.ts",
+        "./src/**/*.mts",
+        "./src/**/*.cts",
+        "test/**/*.ts"
```

Comment:
the test folder may not be always there? e.g if customers disable the test gen or in non-branded we disable test gen by default?